db/utils.go

```diff
@@ -0,0 +1,30 @@
+package db
+
+import (
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+	"golang.org/x/net/context"
+)
+
+type BackgroundTaskFunc func(ctx context.Context) error
+
+// backgroundTask runs task at the specified time interval in its own goroutine until stopped
+func NewBackgroundTask(name string, contextID string, task BackgroundTaskFunc, interval time.Duration, c chan bool) {
+	ctx := context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: base.NewChangeCacheContextID(contextID)})
```

Comment:
As discussed:
- rename NewChangeCacheContextID to NewTaskID
- pass in the task name (along with context id) to replace '-ChangeCache-'
- convert the task names to something a bit less verbose if possible
- move this context creation down to line 20, to instantiate a new id every time the task fires